packages/kbn-migrations/README.md

```diff
@@ -0,0 +1,322 @@
+# Migrations
+
+Kibana migrations provide a systematic and predictable way to upgrade data from one version of Kibana / plugins to a later version. The system has been designed in such a way that upgrading from one version of Kibana to any other should be possible in one go (at least from the standpoint of upgrading documents in the Kibana index).
+
+The core logic for migrations is index / alias (refered to simply as index from here on) agnostic, but it does make an assumption that the index documents are compatible with the Kibana saved-object-client conventions.
+
+
+## How it works
+
+There are two primary use-cases for migrations. The first is migrating all documents in an index from one version to another (e.g. from Kibana 6.4 -> 7.2). The second is migrating documents that are being imported into an index.
```

Comment:
I find this statement somewhat confusing. The main use-case for migrations is changing the mappings or shape of the saved object data. For example; renaming a property or changing it from keyword to text.

There are two sources which we need to deal with, the existing data and data which is imported through the import UI or import/export API